Source/Core/VideoCommon/LightingShaderGen.h

```diff
@@ -62,7 +62,8 @@ static void GenerateLightShader(T& object, LightingUidData& uid_data, int index,
 		case LIGHTATTN_NONE:
 		case LIGHTATTN_DIR:
 			object.Write("ldir = normalize(" LIGHT_POS".xyz - pos.xyz);\n", LIGHT_POS_PARAMS(index));
-			object.Write("attn = 1.0f;\n");
+			object.Write("attn = 1.0;\n");
+			object.Write("if (length(ldir) == 0.0)\n\t ldir = float3(1.0, 1.0, 1.0);\n");
```

Comment:
hmmm what exactly is 0x1p60?  Is that a typo?  So... we aren't actually after a proper normalized ldir here, I doubt we could get one here with the values coming in, basically in the instance of light->pos being at bogus numbers... we just want to use Ambient Lighting.  So either the GC hardware relied on this failure or we aren't setting the proper Attenuation Mode or Diffuse Mode in the channel before we get to the lighting calcs (haven't seen a game set LIGHTATTN_NONE).  This doesn't happen very often, but it does for a minor few things like Mario Tennis.

If we believe that div by infinity or checks for infinity are too costly performance-wise, then I wonder if x, y, or z is > viewport.zfar  or 0xFFFFFF? would be a good enough check here... because that would indicate a light out of range and we can apply just the ambient color to it.